osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -1,36 +1,127 @@
+            private readonly Box background;
+
+            public string TooltipText { get; set; }
+
+            public DifficultyIconTooltip(Color4 accentColour)
+            {
+                AutoSizeAxes = Axes.Both;
+                Masking = true;
+                CornerRadius = 5;
+
+                Children = new Drawable[]
+                {
+                    background = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both
+                    },
+                    new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Vertical,
+                        Padding = new MarginPadding(10),
+                        Children = new Drawable[]
+                        {
+                            difficultyName = new OsuSpriteText
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Font = OsuFont.GetFont(size: 16, weight: FontWeight.Bold),
+                            },
+                            new FillFlowContainer
+                            {
+                                AutoSizeAxes = Axes.Both,
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Direction = FillDirection.Horizontal,
+                                Children = new Drawable[]
+                                {
+                                    starRating = new OsuSpriteText
+                                    {
+                                        Anchor = Anchor.Centre,
+                                        Origin = Anchor.Centre,
+                                        Font = OsuFont.GetFont(size: 16, weight: FontWeight.Regular),
+                                        Colour = accentColour
+                                    },
+                                    new SpriteIcon
+                                    {
+                                        Anchor = Anchor.Centre,
+                                        Origin = Anchor.Centre,
+                                        Margin = new MarginPadding { Left = 4 },
+                                        Icon = FontAwesome.Solid.Star,
+                                        Size = new Vector2(12),
+                                        Colour = accentColour,
+                                    },
+                                }
+                            }
+                        }
+                    }
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                background.Colour = colours.GreyCarmineDark;
+            }
+
+            public void Refresh()
+            {
+                var info = TooltipText.Split('$');
```

Comment:
This is really awkward, but I can see why it was done this way. `TooltipContainer` is a bit awkward.

Reply:
For now will change the separator with another symbol. Should I need framework side change for this?